File: f77/driver/defstate.f
C
        Subroutine Define_Real_State(State_Index,State_Deriv_Index)
C
C       This routine adds an integration to the REAL integrator list.
C
C       Inputs:
C               State_Index         - Location of state in Real_Array [Integer]
C               State_Deriv_Index   - Location of state derivative in Real_Array [Integer]
C
C       Outputs:
C               IxDot_Real - Real_Array locations of state derivatives [Integer Array]
C               Ix_Real    - Real_Array locations of states [Integer Array]
C               Ndes_Real  - Number of real states [Integer]
C
C       Internal variables and constants:
C               None.
C
        Include 'deriv.inc'
        Integer State_Index, State_Deriv_Index
C
        Ndes_Real             = Ndes_Real + 1
        Ix_Real(Ndes_Real)    = State_Index
        IxDot_Real(Ndes_Real) = State_Deriv_Index
C
        Return
        End

File: gcc/driver/defstate.c
#include "../driver/deriv.h"
void define_real_state ( INT8 state_index         // Location of state in Real_Array [Integer]
                       , INT8 state_deriv_index   // Location of state derivative in Real_Array [Integer]
                       )
{
  ndes_real = ndes_real + 1;
  ix_real[ndes_real] = state_index;
  ixdot_real[ndes_real] = state_deriv_index;
}
